### PR TITLE
Support for add-unit in new Kubernetes charms.

### PR DESCRIPTION
**Pull request by tlm**
Add support for caas Application charms to have add-unit work.

## Checklist

 - [x] Requires a [pylibjuju](https://github.com/juju/python-libjuju) change
 - [x] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR
 - [x] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed
 - [x] Comments answer the question of why design decisions were made

## QA steps

Deploy Ben's snappass-test charm using he's guide.

juju add-model snappass

juju add-unit snappass-test 

\#  Verify that Kubernetes has scaled the deployment and that Juju status reports the new scale.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1919421

### CODE REVIEW

**Comment by hpidcock** [comment]
[2.9 - juju add-model failure w 2.8 client and 2.9 controller](https://trello.com/c/BOjnGPKJ/1467-29-juju-add-model-failure-w-28-client-and-29-controller)

**Comment by hpidcock** [comment]
[make juju add-unit work with new pebble style units.](https://trello.com/c/KtpW4R6R/1454-make-juju-add-unit-work-with-new-pebble-style-units)

File: caas/kubernetes/provider/application/scale_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/juju/juju/caas"
+)
+
+func (s *applicationSuite) TestApplicationScaleStateful(c *gc.C) {
```
Comment (hpidcock) [inline caas/kubernetes/provider/application/scale_test.go:17]: May as well add a test for deployment too.

File: caas/kubernetes/provider/scale/scale.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+func PatchReplicasToScale(
+	ctx context.Context,
+	name string,
+	scale int,
+	patcher ScalePatcher) error {
+	if scale < 0 {
+		return errors.NewNotValid(nil, "scale cannot be < 0")
+	}
+
+	patch := fmt.Sprintf("{\"spec\":{\"replicas\": %d}}", scale)
```
Comment (hpidcock) [inline caas/kubernetes/provider/scale/scale.go:70]: Should really use JSON marshal

**Review by hpidcock — APPROVED** [review]
I don't think this is the whole picture when it comes to scale but this is a good first PR on this topic, we can revisit later.

File: caas/kubernetes/provider/scale/scale.go (hunk 2 above)
Comment (tlm) [reply caas/kubernetes/provider/scale/scale.go:70]: can do

File: caas/kubernetes/provider/application/scale_test.go (hunk 1 above)
Comment (tlm) [reply caas/kubernetes/provider/application/scale_test.go:17]: can do

File: caas/kubernetes/provider/scale/scale.go (hunk 2 above)
Comment (tlm) [reply caas/kubernetes/provider/scale/scale.go:70]: Done

**Comment by tlm** [comment]
$$merge$$

**Comment by tlm** [comment]
$$merge$$

**Comment by tlm** [comment]
$$merge$$

**Comment by tlm** [comment]
$$merge$$

**Comment by tlm** [comment]
$$merge$$

**Pull request merged**